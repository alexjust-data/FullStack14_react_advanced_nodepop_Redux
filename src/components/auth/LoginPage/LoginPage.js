import { useDispatch } from 'react-redux';
import { authLogin } from '../../../store/actions.js';

import React from 'react';
import { useLocation, useNavigate } from 'react-router-dom';

import { login } from '../service';
import LoginForm from './LoginForm';
import useMutation from '../../../hooks/useMutation';



function LoginPage() {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const location = useLocation();
  const { isLoading, error, execute, resetError } = useMutation(login);

  const onLogin = () => { 
    dispatch(authLogin());
  };

  const handleSubmit = credentials => {
    execute(credentials)
      .then(() => {
        onLogin(); // Actualiza el estado de autenticaci√≥n en Redux
        const from = location.state?.from?.pathname || '/';
        navigate(from);
      });
  };
  

  return (
    <div>
      <LoginForm onSubmit={handleSubmit} isLoading={isLoading} />
      {isLoading && <p>...login in nodepop</p>}
      {error && (
        <div onClick={resetError} style={{ color: 'red' }}>
          {error.message}
        </div>
      )}
    </div>
  );
}

export default LoginPage;
