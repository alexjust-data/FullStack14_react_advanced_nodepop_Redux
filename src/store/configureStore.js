import { createStore } from 'redux';
import { persistStore, persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage'; 
import rootReducer from './reducers'; 

const persistConfig = {
  key: 'root',
  storage,
  whitelist: ['session'] 
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = createStore(persistedReducer);
export const persistor = persistStore(store);
