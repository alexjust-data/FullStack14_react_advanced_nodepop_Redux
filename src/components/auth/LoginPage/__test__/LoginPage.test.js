import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import LoginPage from '../LoginPage';


jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => jest.fn(),
  useLocation: () => ({
    state: { from: { pathname: '/' } }
  }),
}));

jest.mock('../../context', () => ({
  useAuth: () => ({
    handleLogin: jest.fn(),
  }),
}));

jest.mock('../../service', () => ({
  login: jest.fn(),
}));

jest.mock('../../../../hooks/useMutation', () => () => ({
  isLoading: false,
  error: null,
  execute: jest.fn(),
  resetError: jest.fn(),
}));

const mockStore = configureStore();
const store = mockStore({});

describe('LoginPage', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <MemoryRouter>
          <LoginPage />
        </MemoryRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
