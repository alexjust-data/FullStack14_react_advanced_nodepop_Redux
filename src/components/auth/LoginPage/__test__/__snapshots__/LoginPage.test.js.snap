// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <input
        name="email"
        value=""
      />
      <input
        name="password"
        type="password"
        value=""
      />
      <input
        name="remember"
        type="checkbox"
      />
      <button
        disabled=""
      >
        Login
      </button>
    </form>
  </div>
</DocumentFragment>
`;
